Fix misc stuff (mostly config related)

showell opened the pull request.

rht commented on hosting.md:
````diff
@@ -14,12 +14,17 @@ Here are some convenient options for serving the HTML:
 ```
-jekyll serve --incremental
+gem install jekyll bundler
+jekyll serve
````
Why is the `--incremental` flag gone?

showell replied:
I just wanted the docs to reflect the way I actually tested it.  The original comment didn't explain why you'd use `--incremental` or what the implications are, so I just wanted to keep it simple.  Feel free to submit a PR that fleshes out the Jekyll section a bit here.  You can either stack on top of mine or wait for Tim to merge a few of these commits.

rht replied:
I assume it's because the jekyll server is being run persistently, and it will update itself automatically whenever the archive is being updated (also) incrementally by `archive.py`.

showell replied:
It's still an experimental feature, according to Jekyll.  I assume it works fine for the simple Zulip use case, but I really want to keep the Jekyll docs simple, with just enough pointers to get up and running.

https://jekyllrb.com/docs/configuration/incremental-regeneration/

rht replied:
But eventually people will have to encounter the case when the archive has to be updated. There needs to be a documentation on how this is handled with just `jekyll serve`, without the experimental feature.

showell replied:
`jekyll serve` already looks for updates; it just re-builds the whole archive instead of being smart about it.

If you are interested in making the Jekyll docs better, it's probably a good use of your time to just write up some detailed docs in a readme file and submit a PR.   There's other stuff to explain, like how we make the front matter, etc.

We also want to make sure our solution isn't overly Jekyll-based, since I think a lot of folks would prefer that we just built static HTML that they could serve up directly.

showell replied:
I updated the commit so that it mentions `--incremental` now.

robertylewis replied:
For a Jekyll site that just contains a Zulip archive, I don't think the flag makes much difference, because the "last updated" timestamps force most/all of the pages to be rebuilt each time. In the original leanprover setup, the archive was part of a larger Jekyll site and made up 95% of the Jekyll build time. The `--incremental` flag saved time when we were editing the rest of the site, not so much when we updated the archive.

showell commented:
Just to summarize these changes:

- We now have two major libs: populate.py and html.py
- We have small libs for common stuff and "front matter"
- We have a tests directory (with a few tests)
- We have a settings.py file
- We have at least a "notion" of prod vs. debug
- I fixed two recent regressions in the code.
- I added lots of docstrings throughout the code.

Along the way I made some code cleanups, mostly structural in nature.

rht commented on lib/populate.py:
```diff
@@ -105,32 +105,60 @@ def populate_all(
+
+    streams_data = {}
+
+    for s in streams:
```
Maybe this could be `for s in all_streams`, then have a check
```python
if not is_valid_stream_name(s['name']):
    continue
```

showell replied:
Yeah, I thought about that, but I try to avoid `continue` in big loops when I can, as it's nice to know that there's only one execution path through a loop.

robertylewis commented:
@showell Thank you for all the great work on this! I'm sorry I haven't been paying any attention to this project -- other things have taken priority. Glad to see someone taking the reins. Please feel free to add yourself as an author in the readme if you'd like.
